src/wallet/walletutil.cpp

```diff
@@ -69,7 +75,7 @@ std::vector<fs::path> ListWalletDir()
 
         // Get wallet path relative to walletdir by removing walletdir from the wallet path.
         // This can be replaced by boost::filesystem::lexically_relative once boost is bumped to 1.60.
-        const fs::path path = it->path().string().substr(offset);
+        const fs::path path = it->path().string().substr(offset + 1);
```

Comment:
Suggest we assert substr(0, offset) == what we used for calculating offset above